Sources/REventLogger/REventLoggerCache.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+protocol REventLoggerCacheable {
+    func getTtlReferenceTime() -> Int64
+    func setTtlReferenceTime(_ pushedTimeMs: Int64)
+    func getCurrentTimeInMilliseconds() -> Int64
+}
+
+class EventLoggerCache: REventLoggerCacheable {
+
+    let ttlStorage = UserDefaults.standard
+
+    func getTtlReferenceTime() -> Int64 {
+        if let ttlRefTime = ttlStorage.object(forKey: REventConstants.ttlKey) as? Int64 {
+            return ttlRefTime
+        } else { return -1 }
+    }
+
+    func setTtlReferenceTime(_ pushedTimeMs: Int64) {
+        ttlStorage.set(pushedTimeMs, forKey: REventConstants.ttlKey)
+    }
+
+    func getCurrentTimeInMilliseconds() -> Int64 {
```

Comment:
this can be added in an extension of Date in Utility.
```
extension Date {
    var currentTimeInMS: Int64 {
        Int64(Date().timeIntervalSince1970 * 1000)
    }
}
```